src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -4681,54 +4676,75 @@ void BlockchainLMDB::fixup(fixup_context const context)
 
     try
     {
-      uint64_t num_blocks = height() - start_height;
-      uint64_t prev_cumulative_diff = get_block_cumulative_difficulty(start_height - 1);
-
-      for (size_t i = 0; i < num_blocks; i++)
+      uint64_t const num_blocks                = height() - start_height;
+      uint64_t prev_cumulative_diff            = get_block_cumulative_difficulty(start_height - 1);
+      uint64_t const BLOCKS_PER_BATCH          = 50000;
+      uint64_t const blocks_in_left_over_batch = num_blocks % BLOCKS_PER_BATCH;
+      uint64_t const num_batches               = (num_blocks / BLOCKS_PER_BATCH) + 1; // +1 for leftover blocks
```

Comment:
`(num_blocks + BLOCKS_PER_BATCH - 1) / BLOCKS_PER_BATCH;`
will capture this better (it's basically integer ceil: you won't get the +1 if num_blocks is perfectly divisible by BLOCKS_PER_BATCH).